mq/inmemory/mq.h

```diff
@@ -101,6 +101,13 @@ struct ExportMessageImpl<T_CONSUMER, T_MESSAGE, false> {
 
 template <typename T_CONSUMER, typename T_MESSAGE>
 void ExportMessage(T_CONSUMER& consumer, T_MESSAGE&& message, size_t dropped_count) {
+  // Note that the allowed syntax for `OnMessage()` is:
+  //   either `OnMessage(const T_MESSAGE& message [, size_t dropped_count]);`
+  //   or     `OnMessage(T_MESSAGE&& message [, size_t dropped_count]);`
```

Comment:
Accepting an object by value, when passed in by `std::move()`, would not eliminate the copy, as far as I know.